core/arch/arm/kernel/user_ta.c

```diff
@@ -332,16 +332,16 @@ static TEE_Result user_ta_enter(TEE_ErrorOrigin *err,
 	/* Copy out value results */
 	update_from_utee_param(param, usr_params);
 
+	s = tee_ta_pop_current_session();
+	assert(s == session);
+cleanup_return:
```

Comment:
I don't get it. This label is used only when `tee_mmu_map_param()` fails, in this case why is `tee_mmu_clean_param()` needed? Usually when a function fails, its normal actions do not need to be undone.

Reply:
OK, I'll restore this function and add a cleanup call in `tee_mmu_map_param()`.

Reply:
For "core: user_ta_enter(): clean mapped params" + the fixup commit:

`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`